tests/test_trading_calendar.py

```diff
@@ -575,9 +575,31 @@ def _get_session_block(self):
         # session) block.
 
         if not self.HAVE_EARLY_CLOSES:
-            # If we don't have any early closes, just return a "random" chunk
-            # of three sessions.
-            return self.calendar.all_sessions[10:13]
+            # If we don't have any early closes, return a "random" chunk of
+            # three sessions. To avoid any edge cases, make sure the first and
+            # last sessions are adjacent to days that are also sessions.
+            one_day = pd.Timedelta('1 Day')
+            all_sessions = self.calendar.all_sessions
+            found_enclosed_block = False
+            i = 0
+            while not found_enclosed_block:
+                i += 1
+                try:
+                    block_start = all_sessions[i]
+                    block_end = all_sessions[i + 2]
+                    pre_session = all_sessions[i - 1]
+                    post_session = all_sessions[i + 3]
+                except IndexError:
+                    raise AssertionError(
+                        f'Failed to find a suitable block of test sessions '
+                        f'for the {self.calendar.name} calendar. Does your '
+                        f'test include at least 5 contiguous sessions?'
+                    )
+                if (block_start - one_day == pre_session) and \
+                   (block_end + one_day == post_session):
+                    found_enclosed_block = True
+
+            return all_sessions[i:i + 3]
```

Comment:
This whole change feels a bit hacky, so I'm open to better ideas to fix using "random" dates here.

Reply:
Yeah, I'm not in love with this either. One option might be to just make this a public method to get an "interesting" range of three contiguous dates, and allow tests that need to override it to do so.

Reply:
I considered that, my only qualm was that it defining that method still relied on knowing the "implementation details" so-to-speak of `test_minutes_in_range`, but I'm probably just looking too far into it for a test. Making this a method would probably still be an improvement.

Reply:
I think if you added that method and documented that it should return a range of three contiguous "interesting" dates, that would be sufficient encapsulation.